net-news/snownews/snownews-1.9.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit toolchain-funcs
+
+DESCRIPTION="Command-line RSS feed reader"
+HOMEPAGE="https://github.com/msharov/snownews"
+SRC_URI="https://github.com/msharov/snownews/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86 ~amd64-linux ~x86-linux"
+IUSE=""
+
+COMMON_DEPEND="
+	>=dev-libs/libxml2-2.5.6
+	sys-libs/ncurses:=[unicode(+)]
+	dev-libs/openssl
+	net-misc/curl
+	sys-devel/gettext
+"
+RDEPEND="
+	${COMMON_DEPEND}
+"
+
+DEPEND="
+	${COMMON_DEPEND}
+"
+
+IUSE="debug"
+BDEPEND="virtual/pkgconfig"
+
+src_prepare() {
+	default
+	# Disable stripping in the build system - leave it to the package manager
+	sed -i -e '/ldflags/s/-s/-g -rdynamic/' -e '/cflags/s/-g0/-g/' Config.mk.in || die
+}
+
+src_configure() {
+	econf "$(use_with debug)"
```

Comment:
Does the use flag ```debug``` make any sense? Based on https://github.com/msharov/snownews/blob/master/Config.mk.in, it looks like, that's it's only setting some different (debug) use flags?

Reply:
`configure --with-debug` will set `debug` variable in `Config.mk`, enabling debug build flags, such as `-g`. Without that option the executable is built optimized, with `NDEBUG` define turning off asserts for example. The debug build is only needed for debugging, obviously; most people should use the optimized and stripped executable.

Reply:
That's why I asked. We don't use use flags for setting various build flags, since anyone can via portage set per package own *flags and add things like ```-g```, if he wishes. Only, if debug does also enable more specific parts in the program itself.